src/main.rs

```diff
@@ -585,6 +597,25 @@ fn backtrace(
             .unwrap_or("<unknown>");
         println!("{:>4}: {:#010x} - {}", frame, pc, name);
 
+        // on hard fault exception entry we hit the breakpoint before the subroutine prelude (`push
+        // lr`) is executed so special handling is required
+        // also note that hard fault will always be the first frame we unwind
+        if top_exception.is_none() {
+            top_exception = Some(if pc & !THUMB_BIT == vector_table.hard_fault & !THUMB_BIT {
+                // HardFaultTrampoline
+                println!("      <exception entry>");
+                TopException::HardFault {
+                    // TODO figure how to detect a stack overflow condition
```

Comment:
I guess I could rely on probe-rs::Target info but I think it's possible to select the wrong target and still get the program to flash so relying on that may return a wrong result. Also, the device may have more than one RAM region though that can probably be addressed.